File: src/dataset/Java/triangle_exercism-ThomasZumsteg/openAI/ChatGPT4_Triangle_v3.java
public class Triangle {
    private final double a, b, c;

    public Triangle(double a, double b, double c) throws TriangleException {
        if (a <= 0 || b <= 0 || c <= 0) {
            throw new TriangleException();
        }
        if (a > b) {
            double temp = a;
            a = b;
            b = temp;
        }
        if (b > c) {
            double temp = b;
            b = c;
            c = temp;
        }
        if (a > b) {
            double temp = a;
            a = b;
            b = temp;
        }
        this.a = a;
        this.b = b;
        this.c = c;
        if (this.a + this.b <= this.c) {
            throw new TriangleException();
        }
    }

    public TriangleKind getKind() {
        if (a == c) {
            return TriangleKind.EQUILATERAL;
        } else if (a == b || b == c) {
            return TriangleKind.ISOSCELES;
        } else {
            return TriangleKind.SCALENE;
        }
    }
}

File: src/dataset/python/triangle_exercism-ThomasZumsteg/openAI/ChatGPT4_triangle_v3.py
class TriangleError(Exception):
    def __init__(self, message):
        super().__init__(message)

class Triangle:
    def __init__(self, a, b, c):
        self.a, self.b, self.c = sorted((a, b, c))
        if self.a + self.b <= self.c:
            raise TriangleError("Not a Triangle")

    def kind(self):
        if self.a == self.c:
            return "equilateral"
        if self.a == self.b or self.b == self.c:
            return "isosceles"
        return "scalene"